rholang/src/main/scala/coop/rchain/rholang/interpreter/Registry.scala

```diff
@@ -842,33 +843,37 @@ class RegistryImpl[F[_]](
       case _ => F.unit
     }
 
+  // TODO: Extract hardcoded id:<whatever> and maybe even pass in this resolver itself
+  val blessedContractResolver = HashMap[String, String](
+    "authKey"                 -> "4njqcsc65mt8xrfx9nsz7je7oucgywgd1tj1n1gjtw3ndsophyisro",
+    "basicWallet"             -> "3yicxut5xtx5tnmnneta7actof4yse3xangw4awzt8c8owqmddgyms",
+    "basicWalletFaucet"       -> "r3pfwhwyzfg3n3yhcndwuszkszr11rjdbksizz4eqbqnwg5w49kfo7",
+    "either"                  -> "j6trahbxycumerwpr5qype7j43b7eqh8auebwsa9nn68if47gswh73",
+    "listOps"                 -> "dputnspi15oxxnyymjrpu7rkaok3bjkiwq84z7cqcrx4ktqfpyapn4",
+    "lockbox"                 -> "9dsr55z1js13x346yhhecx66ns486i3yqf6jafrd9p9hdrrbxjqmyu",
+    "makeMint"                -> "exunyijimapk7z43g3bbr69awqdz54kyroj9q43jgu3dh567fxsftx",
+    "makePos"                 -> "nqt875jea4rr83383ys6guzsbebg6k7o7uhrint6d7km67886c4y4s",
+    "nonNegativeNumber"       -> "nd74ztexkao5awjhj95e3octkza7tydwiy7euthnyrt5ihgi9rj495",
+    "pos"                     -> "cnec3pa8prp4out3yc8facon6grm3xbsotpd4ckjfx8ghuw77xadzt",
+    "revVault"                -> "1o93uitkrjfubh43jt19owanuezhntag5wh74c6ur5feuotpi73q8z",
+    "systemInstancesRegistry" -> "wdwc36f4ixa6xacck3ddepmgueum7zueuczgthcqp6771kdu8jogm8",
+    "walletCheck"             -> "oqez475nmxx9ktciscbhps18wnmnwtm6egziohc3rkdzekkmsrpuyt"
+  )
+
   def publicLookup(args: RootSeq[ListParWithRandom], sequenceNumber: Int): F[Unit] =
     args match {
       case Seq(ListParWithRandom(Seq(key, ret), rand)) =>
         def localFail() = fail(ret, rand, sequenceNumber)
         try {
           val Some(Expr(GUri(uri))) = key.singleExpr
           if (uri.startsWith("rho:id:")) {
-            val tail = uri.substring("rho:id:".length)
-            if (tail.size != 54) {
-              localFail()
-            } else {
-              // Could fail
-              // 256 bits plus 14 bit crc-14
-              val bytes: Array[Byte] = ZBase32.decode(tail, 270)
-              val crc: Short =
-                ((bytes(32).toShort & 0xff) | ((bytes(33).toShort & 0xfc) << 6)).toShort
-              if (crc == CRC14.compute(bytes.view.slice(0, 32))) {
-                val args = RootSeq(
-                  ListParWithRandom(
-                    Seq(parByteArray(ByteString.copyFrom(bytes, 0, 32)), ret),
-                    rand
-                  )
-                )
-                lookup(args, sequenceNumber)
-              } else {
-                localFail()
-              }
+            val id = uri.substring("rho:id:".length)
+            handleRhoIdLookup(id, sequenceNumber, ret, rand, localFail _, uri)
+          } else if (uri.startsWith("rho:blessed:")) {
+            val tail = uri.substring("rho:blessed:".length)
```

Comment:
please extract a `val blessedContractPrefix = "rho:blessed:"` (change the name according to the discussion in #rholang https://discordapp.com/channels/391435113415573504/408650938367213568/581384059217903617)